app/controllers/my-projects/assignment/recommendations/add.js

```diff
@@ -274,11 +290,10 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
     const uploadResults = [];
 
     try {
-      // loop through dispositions, uploading all files to each one
       for (let i = 0; i < this.dispositions.length; i += 1) {
         const disposition = this.dispositions.objectAt(i);
 
-        const fileUploadPromises = this.recommendationAddQueue.files.map(file => file.upload(`${ENV.host}/document`, {
+        const fileUploadPromises = this.queuesByDisposition[disposition.id].files.map(file => file.upload(`${ENV.host}/document`, {
```

Comment:
Can there ever be a dispo ID that doesn't have a queue

Reply:
I don't think so, each disposition attached to the assignment should have a file queue I guess. 

@hannahkates would there ever be a disposition which shouldn't have files uploaded to it?